Dampe Fire

serprex opened the pull request:
Particularly important for use with #5435

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3185191071.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3185205830.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3185238165.zip)
<!--- section:artifacts:end -->

serprex committed:
No Dampe Fire

garrettjoecox commented on soh/soh/Enhancements/Cheats/NoDampeFire.cpp:
```diff
@@ -0,0 +1,23 @@
+}
+
+void RegisterNoDampeFire() {
+    COND_ID_HOOK(OnActorInit, ACTOR_EN_HONOTRAP, CVAR_NO_DAMPE_FIRE_VALUE, OnHonoTrapInit);
```
We can prevent the init function from running as well using a `ShouldActorInit` instead of this, but this works fine enough.

garrettjoecox reviewed (approved):
I realize this is a lot more simple than a slider to adjust the difficulty, but I think additionally allowing to go in the other direction to make dampe _harder_ would be more popular among players. Shouldn't block this just my 2 cents.

Malkierian commented:
I would also prefer it to be a difficulty option with a slider, but if you don't want to handle that here, just changing it from CVAR_CHEAT to CVAR_ENHANCEMENT would be enough to get this merged.

serprex commented:
Dampe has 2 variables: a 32 frame cooldown & a 3% chance per frame to drop flame

I expect slider would want to effectively be a multiplier on frequency. So 2x would halve cooldown & double odds of dropping flame. Amusingly this works out so that at 33x dampe will drop a flame 99% of the time every frame. Does that behavior seem suitable for slider?

Pepper0ni commented:
My gut feeling is that it would want more precise control than an integer multiplier, similar to have the frogs timer feels awkward for only having whole number increments, but a float would be too precise and could introduce wierd step functions based on the frame cooldown when set very high (which people will do, because it's funny).

I think handling it with 2 sliders would be clearest: 1 for the cooldown and another for "aggression" (as I think "flame chance per frame" would be misleading with cooldown, we can be honest with how it works in the tooltip)

serprex commented:
2 sliders is excessive. Maybe dropdown inspired by https://github.com/aMannus/Shipwright/pull/9:
* None (0%)
* Normal (32 cd, 3%) (avg 1/3s)
* Jalapeño (10 cd, 3%) (avg 1/2s)
* Chipotle (10 cd, 10%) (avg 1/1s)
* Scotch Bonnet (10 cd, 20%) (avg 2/1s)
* Ghost Pepper (5 cd, 50%) (avg 4/1s)

Pepper0ni commented:
That's fair, but if the highest level isn't "Dampe's Inferno" I'm blocking 😛

Malkierian commented:
If the averages are accurate for them, I'd be OK with that, too. I love the "Dampe's Inferno" idea, lmao.

serprex committed:
spicy

serprex commented:
Implemented. Added Dampe's Inferno as 20/1s

Overall the spicy levels seem alright. The randomness involved dropping the flames really starts to become interesting with Ghost Pepper where you have to weave between flames. I'd want someone to figure out if Dampe's Inferno can be completed in under a minute.. he leaves behind a wall of fire. Maybe using i-frames from rolls

briaguya0 commented:
time for some pepper name bikeshedding!

i'd like to recommend
* None (0%)
* Normal (32 cd, 3%) (avg 1/3s) (i'm not sure about "normal" as the name here. my first other thought was "default" but that doesn't feel perfect either. i just want to make sure it's absolutely clear to players that this is the vanilla value)
* Jalapeño (10 cd, 3%) (avg 1/2s) (great! super well known pepper and [wiki](https://en.wikipedia.org/wiki/Jalape%C3%B1o) has it as 2/5 spicy peppers
* ~Chipotle~ Serrano (10 cd, 10%) (avg 1/1s) (wiki has Chipotle as 2/5 peppers, but [wiki](https://en.wikipedia.org/wiki/Serrano_pepper) has Serrano as 3/5)
* ~Scotch Bonnet~ Habanero (10 cd, 20%) (avg 2/1s) (not sure what scotch bonnet is on wiki but [wiki](https://en.wikipedia.org/wiki/Habanero) has hab at 4/5 and i think it's a well known pepper)
* Ghost Pepper (5 cd, 50%) (avg 4/1s) (this feels perfect to me, and [wiki](https://en.wikipedia.org/wiki/Ghost_pepper) has it at 5/5 as expected)
* Dampe's Inferno (avg 20/1s) love it

serprex commented:
normal -> vanilla?

Scotch bonnet is what I use making sauce, same ballpark as habanero

Modern Jalapeño is 1/5 imo, https://www.dmagazine.com/food-drink/2023/05/why-jalapeno-peppers-less-spicy-blame-aggies

serprex committed:
bikeshed

briaguya0 commented:
i agree jalapeño isn't spicy, just figured going off wiki ratings would give us something to point to.

i was just going for what i think are super well known peppers which is why i suggested hab but I'm fine with whatever

maybe instead of just normal or just default or just vanilla it could be "Default (Vanilla)" - still not certain on that one

JordanLongstaff commented:
I like spiciness, but why not add some milder options between None and Vanilla? Some ideas:
- Banana: (60 cd, 1%) (avg 1/8s)
- Beaver Dam: (50 cd, 2%) (avg 1/5s)

LOL at "Ghost Pepper"

JordanLongstaff commented:
I should also ask, are these cooldowns affected by interpolation?

serprex commented:
> I should also ask, are these cooldowns affected by interpolation?

No

serprex commented:
I don't see value in crowding options between None & Vanilla. There's already a few options more than I'd like

Malkierian commented:
I agree about not wanting to put something between none and vanilla, and I think vanilla by itself as a name is fine, so I'm going to go ahead and merge it.
